backend/DataLib.js

```diff
@@ -1,147 +1,86 @@
+/* eslint-disable no-underscore-dangle */
 /*
  * This file is part of Search NEU and licensed under AGPL3.
  * See the license file in the root folder for details.
  */
 
 import macros from './macros';
 import Keys from '../common/Keys';
-
+import Elastic from './elastic';
 
 // Holds the class data that is used by the search backend
 // and the updater.js
 
-class DataLib {
-  constructor(termDumpMap) {
-    this.termDumpMap = termDumpMap;
-
-
-    // Classes in different terms will never have conflicting hashes
-    // This is a hash map that includes all classes from every term
-    // Used for looking up classes in constant time from any term
-    this.allClassesMap = {};
-
-    this.allSectionsMap = {};
-
-    this.termDumpMap = termDumpMap;
-
-    // Fill up the above two hash maps
-    for (const termDump of Object.values(this.termDumpMap)) {
-      Object.assign(this.allClassesMap, termDump.classMap);
-      Object.assign(this.allSectionsMap, termDump.sectionMap);
-    }
-  }
-
-
-  static loadData(termDumpMap) {
-    const termDumps = Object.values(termDumpMap);
-
-    for (const termDump of termDumps) {
-      if (!termDump.classMap || !termDump.sectionMap) {
-        macros.error('invalid termDump', !!termDumpMap, Object.keys(termDump));
-        return null;
-      }
-    }
-
-
-    return new this(termDumpMap);
-  }
-
-  // Returns a list of the keys in a subject, sorted by classId
-  // Usually takes ~ 5ms and does not instantiate any instances of Class or Subject
-  getClassesInSubject(subject, termId) {
-    if (!this.termDumpMap[termId]) {
-      macros.error("Data lib dosen't have term", termId);
-      return null;
-    }
-
-    const termDump = this.termDumpMap[termId];
-
-    const keys = Object.keys(termDump.classMap);
-
-    const startTime = Date.now();
-
-    const retVal = [];
-    for (const key of keys) {
-      const row = termDump.classMap[key];
-      if (row.subject === subject) {
-        retVal.push(key);
-      }
-    }
-
-    // Sort the classes
-    retVal.sort((a, b) => {
-      return parseInt(termDump.classMap[a].classId, 10) - parseInt(termDump.classMap[b].classId, 10);
-    });
-
-    // Turn this into a analytics call when that is working
-    macros.log('send', 'timing', subject, 'subject', Date.now() - startTime);
-
-    return retVal;
-  }
-
-  getClassesInTerm(termId) {
-    if (!this.termDumpMap[termId]) {
+const DataLib = {
+  getClassesInTerm: async (termId) => {
+    const index = `term${termId}`;
```

Comment:
Can you include the host in this query to ElasticSearch too? The reason we didn't have the host field here before is because there was a different instance of DataLib for each host+termId combination. Like, if in the future we add another college that has the same termIds as one that we already have support for, this could break. tyyyyyyyy :)

Reply:
I'm actually thinking about moving everything to the same index - even if they're different hosts. It will make a lot of the code significantly simpler

Reply:
I know the amount of data we have right now is super small compared to what most people have, (so we'd have good performance either way) but I think it would still be dapper to figure it out and do it the right way - might help if we ever want to add a ton of colleges in the future. I think it would make sense that having separate indices would be better - so when you doing a query against one you don't have to touch the others? 


Here's a link for the other side, but we might find some conflicting stuff on the net lol - note that we would never need to do cross index queries if we did split it across multiple indexes 
https://stackoverflow.com/questions/44700066/elastic-search-one-index-vs-multiple-indexes

Reply:
Did more digging. When we run a query against the single big index with a filter for the host, elasticsearch actually doesn't even look at the documents with the wrong host. It's all optimized at index time when we scraped.

The inverted index that Elasticsearch creates will contain a list of all document ids with neu host. Then when we query for neu classes, it will use that list. Thus, adding more colleges will not affect the performance of the big index - we'll still be using that list, and each college gets their own list. https://www.elastic.co/blog/found-elasticsearch-from-the-bottom-up

Reply:
Ok - lets do it that way :)